### PR TITLE
Manager: clean up outstanding jobs when agents disconnect

### PR DESCRIPTION
**Pull request by sdunnagan**
This is part 1 for addressing issue #99.  This cleans up outstanding jobs
associated with a node when its agent disconnects.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: src/manager/node.c [hunk 1]
```diff
@@ -930,6 +930,25 @@ static int node_disconnected(UNUSED sd_bus_message *message, void *userdata, UNU
...
         if (node->name == NULL) {
                 manager_remove_node(manager, node);
+        } else {
+                /* Remove all jobs associated with the registered node that got
+                   disconnected. */
+                if (!LIST_IS_EMPTY(manager->jobs)) {
+                        Job *job = NULL;
+                        LIST_FOREACH(jobs, job, manager->jobs) {
+                                if (strcmp(job->node->name, node->name) == 0) {
+                                        hirte_log_debugf(
```
Comment (engelmi) [inline src/manager/node.c]: This log seems off. `hirte_log_*` takes care of getting the file/line/func info. I think the first 3 parameters need to be removed.

**Review by engelmi — CHANGES_REQUESTED** [review]

**Comment by engelmi** [comment]
Minor: I think you could combine both commits.

File: src/manager/node.c (hunk 1 above)
Comment (sdunnagan) [reply src/manager/node.c]: `hirte_log_debugf()` doesn't seem to show the file, line and function.

Comment (sdunnagan) [reply src/manager/node.c]: I will remove file/line/func info for purposes of this PR, and I'll find out why I'm not getting those in my print output.

Comment (sdunnagan) [reply src/manager/node.c]: @engelmi solved this issue for me.  I needed to use "stderr-full" in my hirte config in order to get the file/line/func info my prints.  Thanks Michael!

**Review by engelmi — APPROVED** [review]
LGTM

**Pull request merged**